src/shared/parameter/config_definitions/passing_config.yaml

```diff
@@ -79,41 +61,18 @@
      description: >-
        How long we think the enemy will take to recognize we're passing and start
        moving to intercept
- - int:
-     name: num_passes_to_optimize
-     min: 1
-     max: 1000
-     value: 15
-     description: "The number of passes to try to optimize at any given time"
- - int:
-     name: num_passes_to_keep_after_pruning
-     min: 1
-     max: 1000
-     value: 3
-     description: "The number of passes to keep after pruning"
  - int:
      name: number_of_gradient_descent_steps_per_iter
      min: 0
      max: 1000
-     value: 10
+     value: 2 # TODO (#1987) find optimal iterations after tuning, for now 2 does the trick
      description: "The number of steps of gradient descent to perform in each iteration"
  - double:
-     name: pass_equality_max_position_difference_meters
-     min: 0
-     max: 4
-     value: 0.05
+     name: ideal_pass_distance_m
```

Comment:
I agree, this is probably not ideal because sometimes we want to take longer passes if we know enemies can't intercept, will put some more thought into this

This number is currently just a guide to help space out the cherry pickers, we don't use this number to actually look for passes. This value competes with the "progress up field rating" in the rateZone function

Without this, encouraging progress up the field would mean that zone 18 would be ranked highest if the ball was in zone 1. There is nothing "pulling" the cherry pickers back, which is why I added this score.   

Zone selection happens once in every play that passes before a cherrypick->pass->receive cycle. So if we used enemy interception to select the zones and cherry pickers went to those zones, the enemy robots would probably immediately move, potentially invaliding that zone selection.  

But we could simply reassign zones if the pass scores are low for more than a second, which means we can definitely consider enemy proximity and intercept-ability

Ill try to see if this is a quick fix (removing this heuristic and looking at enemy proximity/intercept-abilities) , if its more than a couple lines of change ill create a new ticket

Reply:
I've removed this and added some checks on how the enemy can intercept passes to all 4 corners of the zone + the centre of the zone to rank the zones which seems to work well

removed this heuristic/dynamic param